scripts/check_e2e_tests_are_captured_in_ci_test.py

```diff
@@ -0,0 +1,79 @@
+# Copyright 2020 The Oppia Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Unit tests for scripts/check_e2e_tests_are_captured_in_ci.py."""
+from __future__ import absolute_import  # pylint: disable=import-only-modules
+from __future__ import unicode_literals  # pylint: disable=import-only-modules
+
+import os
+
+from core.tests import test_utils
+import python_utils
+import utils
+
+from . import check_e2e_tests_are_captured_in_ci
+
+TRAVIS_SUITES = [
+    'accessibility', 'additionalEditorFeatures', 'additionalPlayerFeatures',
+    'adminPage', 'collections', 'communityDashboard',
+    'coreEditorAndPlayerFeatures', 'creatorDashboard', 'embedding',
+    'explorationFeedbackTab', 'explorationHistoryTab',
+    'explorationImprovementsTab', 'explorationStatisticsTab',
+    'explorationTranslationTab', 'extensions', 'learner', 'learnerDashboard',
+    'library', 'navigation', 'preferences', 'profileFeatures', 'profileMenu',
+    'publication', 'skillEditor', 'subscriptions', 'topicsAndSkillsDashboard', 'users'
+    ]
+
+
+PROTRACTOR_SUITES = [
+    'accessibility', 'additionalEditorFeatures', 'additionalPlayerFeatures',
+    'adminPage', 'classroomPage', 'collections', 'communityDashboard',
+    'coreEditorAndPlayerFeatures', 'creatorDashboard', 'embedding',
+    'explorationFeedbackTab', 'explorationHistoryTab', 'explorationImprovementsTab',
+    'explorationStatisticsTab', 'explorationTranslationTab', 'extensions', 'fileUploadFeatures',
+    'full', 'learner', 'learnerDashboard', 'library', 'navigation', 'preferences', 'profileFeatures',
+    'profileMenu', 'publication', 'skillEditor', 'subscriptions', 'topicAndStoryEditor',
+    'topicsAndSkillsDashboard', 'users'
+    ]
+
+
+class CheckE2eTestsCapturedInCI(test_utils.GenericTestBase):
```

Comment:
This seems a bit off, actually. I think it would be better to create some test files in core/tests/data or similar and swap out the file locations to use those files instead when testing the code. That way, you can test that the script works correctly (since you can make different sample files to check the success/failure cases), and contributors would not need to modify this file if they add a new e2e test suite. Keep your sample files as small as possible so that any behaviour discrepancies can be easily isolated.

Reply:
Ok Done.